src/main.rs

```diff
@@ -339,6 +382,26 @@ fn cargo_careful(args: env::Args) {
         cmd.args(["--target", target.as_str()]);
     }
 
+    // Enable Main Thread Checker on Apple platforms, as documented here:
+    // <https://developer.apple.com/documentation/xcode/diagnosing-memory-thread-and-crash-issues-early#Detect-improper-UI-updates-on-background-threads>
+    let apple_target = target.contains("-darwin")
+        || target.contains("-ios")
+        || target.contains("-tvos")
+        || target.contains("-watchos");
+    // FIXME: We only do this if the host is running on macOS, since cargo
+    // doesn't currently have a good way of passing environment variables only
+    // to target binaries when using `cargo run` or `cargo test` (and this
+    // means that `rustc` is also run under the main thread checker, which will
+    // probably fail on other platforms).
+    // See <https://github.com/rust-lang/cargo/issues/4505>.
```

Comment:
The comment sounds like it would be doing `cmd.env` next. But then it does something much more targeted? Does `--config env.$name` also get passed to rustc? If yes, then wht is the benefit over `cmd.env`?

Reply:
I've rewritten this comment to be more clear that this is because `DYLD_INSERT_LIBRARIES` is not inherited by child processes (or I think rather, not exposed to the running process' `std::env::vars`).